### PR TITLE
Update version and release docs for release 3.1.2

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update version numbers and release documents for 3.1.2 release.

### CODE REVIEW

File: docs/references/api-reference/C_API.md [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 ## Enums
 
-```{eval-rst}
```
Comment (nightlark) [inline docs/references/api-reference/C_API.md:24]: This change breaks the C API docs page. It needs to be `{eval-rst}` to interpret the block using the rst parser.

Comment (phlptp) [reply docs/references/api-reference/C_API.md:24]: is this the only one of these?  I will change the generation code so it won't happen again.

Comment (phlptp) [reply docs/references/api-reference/C_API.md:24]: I updated it

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,33 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 A note on future revisions.
 Everything within a major version number should be code compatible (with the exception of experimental interfaces). The most notable example of an experimental interface is the support for multiple source inputs. The APIs to deal with this will change in future minor releases. Everything within a single minor release should be network compatible with other federates on the same minor release number. Compatibility across minor release numbers may be possible in some situations but we are not going to guarantee this as those components are subject to performance improvements and may need to be modified at some point. Patch releases will be limited to bug fixes and other improvements not impacting the public API or network compatibility. Check the [Public API](./docs/Public_API.md) for details on what is included and excluded from the public API and version stability.
 
+## [3.1.2][] - 2022-01-25
+
+This patch release primarily to address some build issues on MSYS2, MINGW, and CYGWIN platforms. It also includes some preliminary features for the 3.2 release that were already merged including time monitors, remote logging, and a log buffer.
```
Comment (nightlark) [inline CHANGELOG.md:13]:
```suggestion
This patch release is primarily to address some build issues on MSYS2, MINGW, and CYGWIN platforms. It also includes some preliminary features for the 3.2 release that were already merged including time monitors, remote logging, and a log buffer.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,33 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+
+### Fixed
+
+- Fixed issue building on CYGWIN with the latest FMT library.
+- Build issues related to MSYS2 and Mingw in the networking submodule (submodule updated).
+- Fixed argument processing issue in the comboFed Example.
+
+### Changed
+
+- updates to FMT and SPDLOG
```
Comment (nightlark) [inline CHANGELOG.md:23]:
```suggestion
- Updates to FMT and SPDLOG
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,33 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+- Several documentation updates and cleanup
+- Copyright date changed to 2022
+- Targeted endpoints now allow all method calls, with the restriction being that the destination must be in the target list otherwise an error is generated. An empty destination in methods that allow destinations will operate identically to calls which do not specify a destination.
+- Code cleanup for defining the default port numbers for network core types.
+
+### Added
+
+- A process comms method for federates to instruct HELICS to process messages for a specific amount of wall clock time without granting time or changing state.
+- Added a threaded signal handler to deal some particular issues with killing processes in python and potentially other language interfaces.
```
Comment (nightlark) [inline CHANGELOG.md:32]:
```suggestion
- Added a threaded signal handler to deal with some particular issues with killing processes in python and potentially other language interfaces.
```

File: docs/references/api-reference/C_API.md (hunk 1 above)
Comment (nightlark) [reply docs/references/api-reference/C_API.md:24]: There are several other instances throughout this file that also need to get changed back to `{eval-rst}`.

Comment (nightlark) [reply docs/references/api-reference/C_API.md:24]: There are several other instances throughout this file that also need to get changed back to `{eval-rst}`.

Comment (phlptp) [reply docs/references/api-reference/C_API.md]: ok, think I got it now

**Review by nightlark — APPROVED** [review]

**Pull request merged**